sdk/include/stdlib.h

```diff
@@ -291,17 +291,36 @@ ssize_t __cheri_compartment("allocator")
   heap_quota_remaining(AllocatorCapability heapCapability);
 
 /**
- * Block until the quarantine is empty.
+ * Try to empty the quarantine and defragment the heap.
  *
- * This should be used only in testing, to place the system in a quiesced
- * state.  It can block indefinitely if another thread is allocating and
- * freeing memory while this runs.
+ * This will start revocation sweep and then try to empty the quarantine.  In
+ * normal operation, the allocator will remove a small number of allocations
+ * from quarantine on each allocation.  Allocations are not coalesced until
+ * they are moved from quarantine, so this can cause fragmentation.  If you
+ * have just freed a lot of memory (for example, after resetting a compartment
+ * and calling `heap_free_all`), especially if you have freed a lot of small
+ * allocations, then calling this function will reduce fragmentation.
  *
  * Returns 0 on success, a compartment invocation failure indication
- * (-ENOTENOUGHSTACK, -ENOTENOUGHTRUSTEDSTACK) if it cannot be invoked, or
- * possibly -ECOMPARTMENTFAIL if the allocator compartment is damaged.
+ * (`-ENOTENOUGHSTACK`, `-ENOTENOUGHTRUSTEDSTACK`) if it cannot be invoked, or
+ * possibly `-ECOMPARTMENTFAIL` if the allocator compartment is damaged.
+ *
+ * Returns `-ETIMEDOUT` if the timeout expires or `-EINVAL` if the timeout is
+ * not valid.
+ */
+__attribute__((overloadable)) int __cheri_compartment("allocator")
+  heap_quarantine_flush(Timeout *timeout);
+
+/**
+ * Compatibility implementation of `heap_quarantine_empty`.  This should be
+ * used only in debugging and testing because it is not guaranteed to terminate
+ * if another thread is allocating memory.
  */
-int __cheri_compartment("allocator") heap_quarantine_empty(void);
+static int heap_quarantine_empty(void)
```

Comment:
Could mark this deprecated and add a new version called `heap_quarantine_flush_block_forever`? Hopefully that's a sufficiently scary name to discourage inappropriate use and also consistent with naming convention?

Reply:
We probably could just move this into the test suite, but I don't like breaking existing APIs, even if they're silly.

Reply:
I'd be in favor of moving this to the test suite under a different name and marking this one as deprecated.